widget/richtext.go

```diff
@@ -579,11 +592,19 @@ func (r *textRenderer) layoutRow(texts []fyne.CanvasObject, align fyne.TextAlign
 
 	if realign {
 		for _, text := range texts {
-			if _, ok := text.(*canvas.Text); !ok {
+			txt, ok := text.(*canvas.Text)
+			if !ok {
+				if c, ok := text.(*fyne.Container); ok {
+					wid := c.Objects[0]
+					if link, ok := wid.(*Hyperlink); ok {
+						_, base := fyne.CurrentApp().Driver().RenderedTextSize(link.Text, theme.TextSize(), link.TextStyle)
+						delta := tallestBaseline - base
+						text.Move(fyne.NewPos(c.Position().X, yPos+delta))
+					}
+				}
 				continue
 			}
 
-			txt := text.(*canvas.Text)
 			_, base := fyne.CurrentApp().Driver().RenderedTextSize(txt.Text, txt.TextSize, txt.TextStyle)
```

Comment:
We could avoid doing the same if we save the baselines we got before :), in that case this part could be simply rewritten to:
```go
if realign {
    for i, text := range texts {
        delta := tallestBaseline - baselines[i]
        text.Move(fyne.NewPos(text.Position().X, yPos+delta))
    }
}
...
```

Reply:
I'm not sure if that tidy solution is possible with HyperLink in here too is it?

Reply:
Yes, it is. This code is used for HyperLink:
```go
delta := tallestBaseline - base
text.Move(fyne.NewPos(c.Position().X, yPos+delta))
```
And `c` is the container that was got from `c, ok := text.(*fyne.Container)`, so `c == text`, then the above code is compatible :)

Reply:
Ah yes I see, updated